File: php/src/CnpjFormatter.php
<?php

declare(strict_types=1);

namespace LibCnpj;

final class CnpjFormatter
{
    private const MASK_CHARACTERS = ['.', '/', '-'];
    private const FORMATTED_PATTERN = '/^[A-Z0-9]{2}\.[A-Z0-9]{3}\.[A-Z0-9]{3}\/[A-Z0-9]{4}-[0-9]{2}$/';
    private const ALLOWED_CHARACTERS_PATTERN = '/^[A-Z0-9.\/-]+$/';

    public static function strip(string $value): string
    {
        return str_replace(self::MASK_CHARACTERS, '', $value);
    }

    public static function format(string $value): string
    {
        $clean = self::strip($value);

        if (strlen($clean) !== 14) {
            return $value;
        }

        return sprintf(
            '%s.%s.%s/%s-%s',
            substr($clean, 0, 2),
            substr($clean, 2, 3),
            substr($clean, 5, 3),
            substr($clean, 8, 4),
            substr($clean, 12, 2)
        );
    }

    public static function isFormatted(string $value): bool
    {
        if (preg_match(self::FORMATTED_PATTERN, $value) !== 1) {
            return false;
        }

        return true;
    }

    public static function containsOnlyAllowedCharacters(string $value): bool
    {
        if (preg_match(self::ALLOWED_CHARACTERS_PATTERN, $value) !== 1) {
            return false;
        }

        return true;
    }
}

File: java/src/main/java/com/libcnpj/CnpjFormatter.java
package com.libcnpj;

import java.util.regex.Pattern;

public final class CnpjFormatter {

    private static final Pattern MASK_CHARACTERS = Pattern.compile("[./-]");
    private static final Pattern FORMATTED_PATTERN = Pattern.compile("^[A-Z0-9]{2}\\.[A-Z0-9]{3}\\.[A-Z0-9]{3}/[A-Z0-9]{4}-[0-9]{2}$");
    private static final Pattern ALLOWED_CHARACTERS_PATTERN = Pattern.compile("^[A-Z0-9./-]+$");

    private CnpjFormatter() {
    }

    public static String strip(String value) {
        return MASK_CHARACTERS.matcher(value).replaceAll("");
    }

    public static String format(String value) {
        String clean = strip(value);

        if (clean.length() != 14) {
            return value;
        }

        return clean.substring(0, 2)
            + "."
            + clean.substring(2, 5)
            + "."
            + clean.substring(5, 8)
            + "/"
            + clean.substring(8, 12)
            + "-"
            + clean.substring(12, 14);
    }

    public static boolean isFormatted(String value) {
        return FORMATTED_PATTERN.matcher(value).matches();
    }

    public static boolean containsOnlyAllowedCharacters(String value) {
        return ALLOWED_CHARACTERS_PATTERN.matcher(value).matches();
    }
}
